backend-rust/src/graphql_api.rs

```diff
@@ -3332,26 +3382,151 @@ impl Account {
+            min_index = Some(match min_index {
+                None => statement.id,
+                Some(current_min) => min(current_min, statement.id),
+            });
+
+            max_index = Some(match max_index {
+                None => statement.id,
+                Some(current_max) => max(current_max, statement.id),
+            });
+            connection.edges.push(connection::Edge::new(statement.id.to_string(), statement));
+        }
+
+        if let (Some(page_min_id), Some(page_max_id)) = (min_index, max_index) {
+            let result = sqlx::query!(
+                r#"
+                    SELECT MAX(id) as max_id, MIN(id) as min_id
+                    FROM account_statements
+                    WHERE account_index = $1
+                "#,
+                &self.index
+            )
+            .fetch_one(pool)
+            .await?;
+
+            connection.has_previous_page =
+                result.min_id.map_or(false, |db_min| db_min < page_min_id);
+            connection.has_next_page = result.max_id.map_or(false, |db_max| db_max > page_max_id);
+        }
+
+        Ok(connection)
     }
 
     async fn rewards(
         &self,
-        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: i32,
+        ctx: &Context<'_>,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
-        after: String,
-        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: i32,
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
-        before: String,
+        before: Option<String>,
     ) -> ApiResult<connection::Connection<String, AccountReward>> {
-        todo_api!()
+        let config = get_config(ctx)?;
+        let pool = get_pool(ctx)?;
+        let query = ConnectionQuery::<i64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.reward_connection_limit,
+        )?;
+        let mut rewards = sqlx::query_as!(
+            AccountReward,
+            r#"
+            SELECT
+                id as "id!",
+                block_height as "block_height!",
+                timestamp as "timestamp!",
+                reward_type as "reward_type!: RewardType",
+                amount as "amount!"
+            FROM account_rewards
+            WHERE account_index = $4
+                AND id > $1 AND id < $2
+                ORDER BY id ASC
+                LIMIT $3
+            "#,
+            query.from,
+            query.to,
+            query.limit,
+            &self.index
+        )
+        .fetch(pool);
+
+        let mut connection = connection::Connection::new(false, false);
```

Comment:
I will generalise this in another PR. I would like to have the sorting function included as well